resources/gcloud.rb

```diff
@@ -9,90 +9,284 @@
 end
 
 action :install do
-  platform_cmd = Mixlib::ShellOut.new('uname')
-  platform_cmd.run_command
-  platform_cmd.error!
-  platform = platform_cmd.stdout.strip.downcase
-
-  arch_cmd = Mixlib::ShellOut.new('uname -m')
-  arch_cmd.run_command
-  arch_cmd.error!
-  arch = arch_cmd.stdout.strip
-
-  version = new_resource.version
-  if version.empty?
+  # Check infomation of operation system
+  platform = attr_platform
```

Comment:
Refer ruby convention (https://github.com/bbatsov/ruby-style-guide#classes--modules) it mentioned:
`For accessors and mutators, avoid prefixing method names with get_ and set_. It is a Ruby convention to use attribute names for accessors (readers) and attr_name= for mutators (writers).`
...
also mentioned `Avoid the use of attr.` 
attr_ isn't really good, because it refer to the class attributes of a class

This is one the the convention example ```# bad
class Person
  def get_name
    "#{@first_name} #{@last_name}"
  end

  def set_name(name)
    @first_name, @last_name = name.split(' ')
  end
end

# good
class Person
  def name
    "#{@first_name} #{@last_name}"
  end

  def name=(name)
    @first_name, @last_name = name.split(' ')
  end
end```

According to it, we can just use the name of the variable as name of class, (e.g def platform)